File: read_pdf.py
import pdfplumber

def extract_text(pdf_path):
    text = ""
    with pdfplumber.open(pdf_path) as pdf:
        for page in pdf.pages:
            t = page.extract_text()
            if t:
                text += t + "\n"
    return text

if __name__ == '__main__':
    text = extract_text('public/docs/examquestions.pdf')
    print(text[:2000])

File: read-pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

let dataBuffer = fs.readFileSync('public/docs/examquestions.pdf');

pdf(dataBuffer).then(function(data) {
    console.log(data.text.substring(0, 2000)); // Log the first 2000 characters
});
